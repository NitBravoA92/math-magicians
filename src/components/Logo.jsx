import mathLogo from '../mathLogo.gif';
const Logo = () => (
  <a href='/' id="logo-container">
    <img src={mathLogo} className="App-logo" alt="logo" />
    <span id="main-title">Math Magicians</span>
  </a>
);

export default Logo;