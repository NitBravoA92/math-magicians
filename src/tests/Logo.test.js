import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { render } from "@testing-library/react";
import renderer from 'react-test-renderer';
import Logo from "../components/Logo";

describe("The Logo component", () => {

  test("renders correctly into the DOM", () => {
    const logoComponent = renderer
    .create(<Logo />)
    .toJSON();
    expect(logoComponent).toMatchSnapshot();
  });
  
  test("should display the text: Math Magicians", () => {
    const component = render(<Logo />);
    expect(component.container).toHaveTextContent("Math Magicians");
  });
  
  test("should display a Logo image", () => {
    const component = render(<Logo />);
    const image = component.container.querySelector("img");
    expect(image).toHaveAttribute("src", "mathLogo.gif");
    expect(image).toHaveAttribute("alt", "logo");
  });

});

