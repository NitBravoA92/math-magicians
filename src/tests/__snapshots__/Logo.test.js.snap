// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Logo component renders correctly into the DOM 1`] = `
<a
  className="logo-container"
  href="/"
>
  <img
    alt="logo"
    className="App-logo"
    src="mathLogo.gif"
  />
  <span
    className="logo-title"
  >
    Math Magicians
  </span>
</a>
`;
